server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/draft/model/MailboxFactory.java

```diff
@@ -86,18 +98,55 @@ public MailboxBuilder usingPreloadedMailboxesMetadata(Optional<List<MailboxMetaD
         }
 
         public Optional<Mailbox> build() {
-            Preconditions.checkNotNull(id);
             Preconditions.checkNotNull(session);
 
             try {
-                MessageManager mailbox = mailboxFactory.mailboxManager.getMailbox(id, session);
-                return Optional.of(mailboxFactory.fromMessageManager(mailbox, userMailboxesMetadata, quotaLoader, session));
+                MailboxId mailboxId = computeMailboxId();
+                Mono<MessageManager> mailbox = mailbox(mailboxId).cache();
+
+                MailboxACL mailboxACL = mailboxMetaData.map(MailboxMetaData::getResolvedAcls)
+                    .orElseGet(Throwing.supplier(() -> retrieveCachedMailbox(mailboxId, mailbox).getResolvedAcl(session)).sneakyThrow());
+
+                MailboxPath mailboxPath = mailboxMetaData.map(MailboxMetaData::getPath)
+                    .orElseGet(Throwing.supplier(() -> retrieveCachedMailbox(mailboxId, mailbox).getMailboxPath()).sneakyThrow());
+
+                MailboxCounters mailboxCounters = mailboxMetaData.map(MailboxMetaData::getCounters)
```

Comment:
Can you extract the method to retrieve the mailboxACL, mailboxPath, mailboxCounters from the metadata?

Reply:
I dont understand what you ask for...